src/sbvr-api/sbvr-utils.coffee

```diff
@@ -723,6 +734,39 @@ exports.getAbstractSqlModel = getAbstractSqlModel = (request) ->
 	request.abstractSqlModel ?= abstractSqlModels[request.vocabulary]
 	return request.abstractSqlModel
 
+exports.getAffectedIds = ({ req, request, tx }) ->
+	# We reparse to make sure we get a clean odataQuery, without permissions already added
+	# And we use the request's url rather than the req for things like batch where the req url is ../$batch
+	uriParser.parseOData({
+		method: request.method
+		url: "/#{request.vocabulary}#{request.url}"
+	})
+	.then (request) ->
+		abstractSqlModel = getAbstractSqlModel(request)
+		resourceName = resolveSynonym(request)
+		resourceTable = abstractSqlModel.tables[resourceName]
+		if !resourceTable?
+			throw new Error('Unknown resource: ' + request.resourceName)
+		idField = resourceTable.idField
+
+		_.set(request.odataQuery, [ 'options', '$select' ], { 'properties': [ { 'name': idField } ] })
+
+		permissions.addPermissions(req, request)
+
+		request.method = 'GET'
```

Comment:
Would it make sense to also clear the `body` and the `$expand` & `$orderBy` options of the request?

Reply:
The `$orderby` needs to be kept for cases like `$orderby=name&$top=10` where it affects which ids are affected.  `$expand` can be dropped but I think it's invalid on anything other than a `GET` so shouldn't be able to be here - I need to double check that though and also add a custom case for passing a `GET` request to here (which should never happen imo, but if it does it should probably either throw an error or return `[]` as a GET doesn't affect things)

Reply:
Just-an-idea: How about dropping the `$orderby` when neither `$top` nor `$skip` is used in the current request.
I might be missing an edge case and this might be premature optimization though.

Reply:
Added the check for passing in a GET request and also removed `$expand`s.  I'd say we should just leave $orderby in as the performance difference should be neglible